src/common/libsubprocess/server.c

```diff
@@ -291,15 +311,25 @@ static void proc_output_cb (flux_subprocess_t *p, const char *stream)
         goto error;
     }
 
-    if (len) {
-        if (proc_output (p, stream, s, request, buf, len, false) < 0)
-            goto error;
-    }
-    else {
-        if (proc_output (p, stream, s, request, NULL, 0, true) < 0)
-            goto error;
+    if (!p->bg) {
+        if (len) {
+            if (proc_output (p, stream, s, request, buf, len, false) < 0)
+                goto error;
+        }
+        else {
+            if (proc_output (p, stream, s, request, NULL, 0, true) < 0)
+                goto error;
+        }
     }
+    else if (len) {
+        /* background process: log output (ignore EOF) */
 
+        const char *command = basename_simple (flux_cmd_arg (p->cmd, 0));
+        if (streq (stream, "stderr"))
+            llog_error (s, "%s: %s", command, buf);
+        else
+            llog_info (s, "%s: %s", command, buf);
+    }
```

Comment:
What happens if the user sets the STDIO_FALLTHROUGH local_flag?

Reply:
The stdout will fall through as documented.